crates/bitwarden-core/src/key_management/account_cryptographic_state.rs

```diff
@@ -0,0 +1,497 @@
+    #[error("A generic cryptographic error occurred: {0}")]
+    GenericCrypto(CryptoError),
+}
+
+impl From<CryptoError> for AccountCryptographyInitializationError {
+    fn from(err: CryptoError) -> Self {
+        AccountCryptographyInitializationError::GenericCrypto(err)
+    }
+}
+
+/// Any keys / cryptographic protection "downstream" from the account symmetric key (user key).
+/// Private keys are protected by the user key.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Enum))]
+#[cfg_attr(feature = "wasm", derive(Tsify), tsify(into_wasm_abi, from_wasm_abi))]
+#[allow(clippy::large_enum_variant)]
+pub enum WrappedUserAccountCryptographicState {
+    /// A V1 user has only a private key.
+    V1 {
+        /// The user's encryption private key, wrapped by the user key.
+        private_key: EncString,
+    },
+    /// A V2 user has a private key, a signing key, a signed public key and a signed security state.
+    /// The SignedPublicKey ensures that others can verify the public key is claimed by an identity
+    /// they want to share data to. The signed security state protects against cryptographic
+    /// downgrades.
+    V2 {
+        /// The user's encryption private key, wrapped by the user key.
+        private_key: EncString,
+        /// The user's public-key for the private key, signed by the user's signing key.
+        /// Note: This is optional for backwards compatibility. After a few releases, this will be
+        /// made non-optional once all clients store the response on sync.
+        signed_public_key: Option<SignedPublicKey>,
+        /// The user's signing key, wrapped by the user key.
+        signing_key: EncString,
+        /// The user's signed security state.
+        security_state: SignedSecurityState,
+    },
+}
+
+impl WrappedUserAccountCryptographicState {
+    /// Converts to a PrivateKeysResponseModel in order to make API requests. Since the
+    /// [WrappedUserAccountCryptographicState] is encrypted, the key store needs to contain the
+    /// user key required to unlock this state.
+    pub fn to_private_keys_request_model(
+        &self,
+        store: &KeyStore<KeyIds>,
+    ) -> Result<PrivateKeysResponseModel, AccountCryptographyInitializationError> {
+        let verifying_key = self.verifying_key(store)?;
+        Ok(PrivateKeysResponseModel {
+            object: Some("privateKeys".to_string()),
+            signature_key_pair: match self {
+                WrappedUserAccountCryptographicState::V1 { .. } => None,
+                WrappedUserAccountCryptographicState::V2 { signing_key, .. } => Some(Box::new(
+                    bitwarden_api_api::models::SignatureKeyPairResponseModel {
+                        wrapped_signing_key: Some(signing_key.to_string()),
+                        verifying_key: Some(
+                            B64::from(
+                                verifying_key
+                                    .clone()
```

Comment:
You should be able to use `.as_ref()` here to avoid the clone.
```suggestion
                                    .as_ref()
```